### PR TITLE
Refactor linting errors

### LINKED ISSUE #8: createOSPKG adds additional `zip` prefix
We should probably only add a zip prefix, if the ospkg does not contain it already.

```shell
task: [demo:ospkg] mkdir -p out/ospkgs
task: [demo:ospkg] stmgr createOSPKG -out 'out/ospkgs/os-pkg-example-ubuntu20.zip' -label='System Transparency Test OS' -kernel=cache/debos/ubuntu-focal-amd64.vmlinuz -initramfs=cache/debos/ubuntu-focal-amd64.cpio.gz -cmdline='console=tty0 console=ttyS0,115200n8 rw rdinit=/lib/systemd/systemd' -url=http://10.0.2.2:8080/os-pkg-example-ubuntu20.zip
task: [demo:ospkg] for i in {1..2}; do stmgr sign -key=out/keys/example_keys/signing-key-$i.key -cert=out/keys/example_keys/signing-key-$i.cert -ospkg out/ospkgs/os-pkg-example-ubuntu20.zip; done
[demo:ospkg] ERROR: Runtime error: open out/ospkgs/os-pkg-example-ubuntu20.zip: no such file or directory
task: Failed to run task "demo:ospkg": exit status 1
$ ls out/ospkgs/
os-pkg-example-ubuntu20.zip.json  os-pkg-example-ubuntu20.zip.zip
```

### PR DESCRIPTION
**Pull request by MDr164**
Code refactoring and resolving of most linting errors. Also fixes https://github.com/system-transparency/stmgr/issues/8

### CODE REVIEW

File: keygen/certificate.go [hunk 1]
```diff
@@ -16,77 +17,98 @@ const (
...
-		newCert, newKey, err = newCertWithED25519Key(rootCert, &rootKey, notBefore, notAfter)
-		if err != nil {
-			return err
-		}
+	return writeToDisk(newCert, newKey, keyOut, args.CertOut)
+}
+
+func checkArgs(args *Args) error {
+	if args.IsCa && (len(args.RootCertPath) != 0 || len(args.RootKeyPath) != 0) {
+		return errors.New("isCa specified, will ignore rootKey and rootCert")
```
Comment (jenstopp) [inline keygen/certificate.go]: Either return no error here or reword the message

File: keygen/certificate.go [hunk 2]
```diff
@@ -16,77 +17,98 @@ const (
 	DefaultRootCertName = "rootcert.pem"
 	DefaultKeyName      = "key.pem"
 	DefaultRootKeyName  = "rootkey.pem"
+	DefaultDuration     = 72 * time.Hour
```
Comment (jenstopp) [inline keygen/certificate.go]: Could be a bit more precise if set as an exported constant. Maybe "DefaultValidPeriod"
Or declare it private right above the function. Do we need to expose this?

File: logging/logging_test.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package logging_test
+
+import (
+	"testing"
+)
+
+func TestNewLogger(t *testing.T) {
+	// TODO
```
Comment (jenstopp) [inline logging/logging_test.go]: Interesting… Why doesn't the linter complain about TODO in the other test.go files

File: keygen/util.go [hunk 4]
```diff
@@ -3,9 +3,12 @@ package keygen
 import (
 	"encoding/pem"
 	"errors"
+	"io/fs"
 	"os"
 )
 
+const defaultFilePerm fs.FileMode = 0o600
```
Comment (jenstopp) [inline keygen/util.go:15]: Why not `os.ModePerm` ?

File: ospkg/create.go [hunk 5]
```diff
@@ -1,13 +1,36 @@
...
 import (
+	"io/fs"
 	"os"
 
 	ospkgs "github.com/system-transparency/stboot/ospkg"
 )
 
-func Create(out, label, url, kernel, initramfs, cmdline string) error {
-	osp, err := ospkgs.CreateOSPackage(label, url, kernel, initramfs, cmdline)
+const defaultFilePerm fs.FileMode = 0o600
```
Comment (jenstopp) [inline ospkg/create.go:10]: Why not `os.ModePerm`?

File: ospkg/create_test.go [hunk 6]
```diff
@@ -1,9 +1,13 @@
-package ospkg
+package ospkg_test
```
Comment (jenstopp) [inline ospkg/create_test.go:1]: Why is it a different package?

File: ospkg/sign_test.go [hunk 7]
```diff
@@ -0,0 +1,11 @@
+package ospkg_test
```
Comment (jenstopp) [inline ospkg/sign_test.go:1]: Why is it a different package?

File: provision/cfgtool_test.go [hunk 8]
```diff
@@ -1,9 +1,13 @@
-package provision
+package provision_test
```
Comment (jenstopp) [inline provision/cfgtool_test.go]: See above

File: provision/parser.go [hunk 9]
```diff
@@ -2,12 +2,15 @@ package provision
...
 import (
 	"encoding/json"
+	"io/fs"
 	"os"
 
 	guid "github.com/google/uuid"
 	"github.com/system-transparency/efivar/efivarfs"
 )
 
+const defaultFilePerm fs.FileMode = 0o600
```
Comment (jenstopp) [inline provision/parser.go:12]: Why not `os.ModePerm`

File: provision/parser_test.go [hunk 10]
```diff
@@ -0,0 +1,7 @@
+package provision_test
```
Comment (jenstopp) [inline provision/parser_test.go]: See above

File: provision/tui_test.go [hunk 11]
```diff
@@ -0,0 +1,51 @@
+package provision_test
```
Comment (jenstopp) [inline provision/tui_test.go]: See above

**Review by jenstopp — COMMENTED** [review]
I like that linting stuff. However, let's make it 100% happy then or disable some tests.

File: keygen/util.go (hunk 4 above)
Comment (MDr164) [reply keygen/util.go:15]: os.ModePerm is equal to 0777 and not 0600 which is a bit too lax I'd  say. But if 0777 is fine for the project I'll switch it around.

File: ospkg/create.go (hunk 5 above)
Comment (MDr164) [reply ospkg/create.go:10]: See above

File: ospkg/create_test.go (hunk 6 above)
Comment (MDr164) [reply ospkg/create_test.go:1]: Can't give a clever explanation other than the new linter wanted it that way and the IDE also recommends it now with the replaced linter. The compiler also does not complain so I guess it was meant to place test code under "packagename_test" in the same folder even if that string isn't identical to "packagename"

File: ospkg/sign_test.go (hunk 7 above)
Comment (MDr164) [reply ospkg/sign_test.go:1]: See above

File: provision/cfgtool_test.go (hunk 8 above)
Comment (MDr164) [reply provision/cfgtool_test.go]: See above

File: provision/parser.go (hunk 9 above)
Comment (MDr164) [reply provision/parser.go:12]: See above

File: provision/parser_test.go (hunk 10 above)
Comment (MDr164) [reply provision/parser_test.go]: See above

File: provision/tui_test.go (hunk 11 above)
Comment (MDr164) [reply provision/tui_test.go]: See above

File: logging/logging_test.go (hunk 3 above)
Comment (MDr164) [reply logging/logging_test.go]: It doesn't, there are just so many that it usually only complains about a few. Will remove them as it is obvious the tests are missing and will be added in the next PR.

File: keygen/certificate.go (hunk 1 above)
Comment (MDr164) [reply keygen/certificate.go]: Thought the same right after pushing the commits. Gonna change those to warnings instead.

File: ospkg/create_test.go (hunk 6 above)
Comment (jenstopp) [reply ospkg/create_test.go:1]: In `pkgname_test`, can you still access unexpected things from `pkgname`?

Comment (jenstopp) [reply ospkg/create_test.go:1]: https://stackoverflow.com/questions/19998250/proper-package-naming-for-testing-with-the-go-language

I think we should disable this test. From my point of view, white-box testing is worth it, and we should be able to write the tests in the same package. As pointed above, there is nothing wrong with it.

File: ospkg/sign_test.go (hunk 7 above)
Comment (MDr164) [reply ospkg/sign_test.go:1]: Yes I agree, checked the docs and they also don't explicitly recommend using the `_test` package name. Gonna revert this change then and disable the linting test that checks those.

File: keygen/certificate.go (hunk 1 above)
Comment (MDr164) [reply keygen/certificate.go]: Done

File: keygen/certificate.go (hunk 2 above)
Comment (MDr164) [reply keygen/certificate.go]: Done

File: logging/logging_test.go (hunk 3 above)
Comment (MDr164) [reply logging/logging_test.go]: Done

File: ospkg/create_test.go (hunk 6 above)
Comment (MDr164) [reply ospkg/create_test.go:1]: Yes I agree, checked the docs and they also don't explicitly recommend using the _test package name. I reverted this change and disabled the linting test that checks those.

**Review by jenstopp — APPROVED** [review]

**Pull request merged**